src/frontend/admin/src/components/presentational/table/TableComponent.tsx

```diff
@@ -27,9 +29,12 @@ interface Props<T> {
   topActions?: React.ReactElement;
 }
 
-export function TableComponent<T>({ enableEdit = true, ...props }: Props<T>) {
+export function TableComponent<T extends GridValidRowModel>({
+  enableEdit = true,
+  paginationMode = "server",
```

Comment:
We always want to use server pagination mode nope ?

Reply:
I'm not sure about that, I prefer to leave the choice and for the component to be as generic as possible. Otherwise I would also have added the notion of useTableResourcePagination directly inside the component.